src/directLine.ts

```diff
@@ -949,7 +950,21 @@ export class DirectLine implements IBotConnection {
             ws.onclose = close => {
                 konsole.log("WebSocket close", close);
                 if (sub) sub.unsubscribe();
-                subscriber.error(close);
+
+                // RxJS.retryWhen has a bug that would cause "error" signal to be sent after the observable is completed/errored.
+                // We need to guard against extraneous "error" signal to workaround the bug.
+                closed || subscriber.error(close);
```

Comment:
>error [](start = 37, length = 5)

normal web socket close should lead to non-error RX stream completion, right?

Reply:
I am keeping the original logic as-is, i.e. `onclose` will throw an error. And I think it's intentional.

If you look at line 910, https://github.com/microsoft/BotFramework-DirectLineJS/blob/6fc932dd1e3c4ad11fb84db02bd0d47ade987dd3/src/directLine.ts#L912, the comment said: 

```
this.observableWebSocket<ActivityGroup>()
  // WebSockets can be closed by the server or the browser. In the former case we need to
  // retrieve a new streamUrl. In the latter case we could first retry with the current streamUrl,
  // but it's simpler just to always fetch a new one.
  .retryWhen(error$ => error$.delay(this.getRetryDelay(), this.services.scheduler).mergeMap(error => this.reconnectToConversation()))
```

Looks like the author of the logic intentionally want to retry whenever the server or browser close the connection. And to do that, the code need to throw error on graceful close.

What do you think?